src/ott/geometry/costs.py

```diff
@@ -139,6 +140,16 @@ def pairwise(self, x: jnp.ndarray, y: jnp.ndarray) -> float:
     """Compute cost as evaluation of :func:`h` on :math:`x-y`."""
     return self.h(x - y)
 
+  def tree_flatten(self):
+    """Tree flatten."""
+    return (), (self.p,)
+
+  @classmethod
+  def tree_unflatten(cls, aux_data, children):
+    """Tree unflatten."""
+    del children
+    return cls(aux_data[0])
```

Comment:
I moved these up in the class hierarchy, not sure if desired, tests seem to pass locally.

Reply:
But there's no need to document these.